NMS-14171: add docs for event performance data (#4637)
OpenNMS/opennms: docs/modules/operation/pages/events/perf-data.adoc

= Performance data

This feature initially is triggered by customer who want to persist SNMP traps in time series database. Since traps are already persist in OpenNMS an events form. Moreover, trap output is not the same as SNMP. It always embeds values into a string. Events support named regex extraction, and it is a good starting point for persisting extracted parameters into time series.
== Flow of persisting

> - event matching with eventconf and check collection information exists

Author: It refer to trap's output data.

> - It will try to lookup parameters stated in config against event's parameters

Author: "It" is the feature, or you could say OpenNMS.

> - if paramValue in eventconf exists, it will convert the mating key into numeric values.

Author: It is a typo.

> - Based on the collection setting, it will persist data into nodeSnmp, interfaceSnmp or resource type.

Author: It means the feature / OpenNMS

 DEF:octIn={rrd1}:bytesIn:AVERAGE \
 AREA:octIn#73d216: \
 LINE1:octIn#4e9a06:"In " \
 GPRINT:octIn:AVERAGE:"Avg  \\: %8.2lf %s" \
 GPRINT:octIn:MIN:"Min  \\: %8.2lf %s" \
 GPRINT:octIn:MAX:"Max  \\: %8.2lf %s\\n"

report.VALUE.name=VALUE
report.VALUE.columns=VALUE
report.VALUE.type=eventType
report.VALUE.command=--title="VALUE" \
 --vertical-label="Bytes per second" \
 DEF:octOut={rrd1}:bytesOut:AVERAGE \
 AREA:octOut#73d216: \
 LINE1:octOut#4e9a06:"Out " \
 GPRINT:octOut:AVERAGE:"Avg  \\: %8.2lf %s" \

report.uei.opennms.org_traps_test_varbind.name=uei.opennms.org_traps_test_varbind
report.uei.opennms.org_traps_test_varbind.columns=uei.opennms.org_traps_test_varbind
report.uei.opennms.org_traps_test_varbind.type=interfaceSnmp
report.uei.opennms.org_traps_test_varbind.command=--title="uei.opennms.org_traps_test_varbind" \
 --vertical-label="Bytes per second" \
 DEF:octIn={rrd1}:bytesIn:AVERAGE \
 AREA:octIn#73d216: \
 LINE1:octIn#4e9a06:"In " \
 GPRINT:octIn:AVERAGE:"Avg  \\: %8.2lf %s"
```

#### create resource-types if you are using generic type
If your time series data type is non nodeSnmp / interfaceSnmp. You will need to create a resource type file in etc/resource-types.d.

> The most important thing is to match the name with evenconf's resourceType. There is also ${instance} variable available.

Reviewer: ```suggestion
Make sure you match the name with evenconf's resourceType. 
There is also ${instance} variable available.
```
why is the {instance} variable important?

Author: for example you are receiving a network switch traps. {instance} can be port name. Without a proper name, you won't know what is that events bind to.
